modin/data_management/functions/groupby_function.py

```diff
@@ -29,9 +30,13 @@ def caller(
             numeric_only=True,
             drop=False,
         ):
-            assert isinstance(
-                by, type(query_compiler)
-            ), "Can only use groupby reduce with another Query Compiler"
+            if not isinstance(by, type(query_compiler)):
```

Comment:
What cases this is working now for?

Reply:
It works for all cases we couldn't handle in `DataFrame.groupby` and transform `by` to a `QueryCompiler`. The case of our interest is NYC Taxi query like the following:
```
df = df.groupby(["b", df["c"].dt.year]).size()
```

Reply:
I see, thanks!

Reply:
I see, thanks!